tests/unittests/DeviceManagerTest.cpp

```diff
@@ -481,6 +486,86 @@ TEST_P(DeviceManagerTest, ReuseModule) {
   EXPECT_TRUE(result2->isEqual(output2));
 }
 
+TEST(DeviceManagerTest, SetDeviceMemory) {
+
+  // Never set a value.
+  auto configEmpty = llvm::make_unique<DeviceConfig>(BackendKind::CPU);
+  EXPECT_EQ(configEmpty->getDeviceMemory(), 0);
+  // Has set a value.
+  auto configFull = llvm::make_unique<DeviceConfig>(BackendKind::CPU);
+  configFull->setDeviceMemory(32768);
+  EXPECT_EQ(configFull->getDeviceMemory(), 32768);
+
+#ifdef GLOW_WITH_CPU
```

Comment:
Can this be done as TEST_P? look at other examples how that could help handling different backend kinds.
I'd prefer to limit usage of #ifdefs here and there.

Reply:
> Can this be done as TEST_P? look at other examples how that could help handling different backend kinds.
> I'd prefer to limit usage of #ifdefs here and there.

I find that the test logic of those different backends are little bit different. To remove the `#ifdef`s, I moved them to different test files.  Should I merge the CPU and Interpreter test code into one TEST_P?

Reply:
> Can this be done as TEST_P? look at other examples how that could help handling different backend kinds.
> I'd prefer to limit usage of #ifdefs here and there.

I've looked through all backend tests. I think to merge those tests into one parameterized (TEST_P), there are a lot of extra functions required to implement. 
For example, the CPU and Interpreter backend will read the command line option as their default memory, which is not exposed to our test. The default createdevicemanager() does not accept a default memory size as its parameter. Also, those different backends have different default memory size, so the test should verify separately. The CPU and interpreter backend should prefer command line option rather than the deviceConfig. On the other side, the OpenCL backend will check the actual device memory size using OpenCL API.  If it is larger than deviceConfig memory size, we should prefer to use the size limited by deviceConfig. 
The habana default memory size is in KB and others are in Bytes.  Overall, to merge them into one Test_P, we still need to write may branches for different backend anyway without implementing a lot of extra helper functions. Could we just stay with current Test for now (I've moved those tests to different test file for different backends. So the #ifdefs have been removed) and maybe merge them later?